File: з4.pas
program z4;
var year,minYear, maxYear, i: integer;
  kolvoos, // количество осадков для нынешного года
  minos,     // минимальное колво осадков 
  maxos: real; // максимальное колво осадков 
begin
  writeln('Введите количество осадков в мм:');

  write('2015 год: ');
  readln(kolvoos);
  
  minos := kolvoos; 
  maxos := kolvoos; 
  minYear := 2015;  
  maxYear := 2015;  

  for i := 1 to 10 do
  begin
    year := 2015 + i; 
    write(year, ' год: ');

    readln(kolvoos);
    
    if kolvoos < minos then
    begin
      minos := kolvoos; // обновляю значение минимума
      minYear := year;
    end;
    
    if kolvoos > maxos then
    begin
      maxos:= kolvoos; // обновляю значение максимума
      maxYear := year;
    end;
  end;
  
  writeln;

  writeln('самый засушливый год: ', minYear, ' (', minos:0:1, ' мм)');
  writeln('самый влажный год: ', maxYear, ' (', maxos:0:1, ' мм)');
end.

File: з4.py
print('введите количество осадков в мм:')

print('2015 год: ')
kolvoos = float(input())

minos = kolvoos
maxos = kolvoos
minYear = 2015
maxYear = 2015

for i in range(1, 11):
    year = 2015 + i
    print(year, ' год: ')

    kolvoos = float(input())

    if kolvoos < minos:
        minos = kolvoos
        minYear = year

    if kolvoos > maxos:
        maxos = kolvoos
        maxYear = year

print('самый засушливый год: ', minYear, ' (', format(minos, '.1f'), ' мм)')
print('самый влажный год: ', maxYear, ' (', format(maxos, '.1f'), ' мм)')
